manifests/install.pp

```diff
@@ -19,7 +19,13 @@
         extract_path => $vault::bin_dir,
         source       => $vault::real_download_url,
         cleanup      => true,
-        creates      => $vault_bin,
+        creates      => $facts['vault_version'] ? { # lint:ignore:selector_inside_resource
```

Comment:
this while block is a bit hard to read, can you add a test for it?

Reply:
How would you do it?
I presume the fact will be nil on tests, since fact will be loaded before tests.

Reply:
I added the tests, could you check if they fit?